entity-framework/core/querying/sql-queries.md

```diff
@@ -0,0 +1,163 @@
+
+> [!NOTE]
+>
+> <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> can only be used directly on a `DbSet`. It cannot be composed over an arbitrary LINQ query.
+
+## Passing parameters
+
+> [!WARNING]
+> **Pay close attention to parameterization when using SQL queries**
+>
+> When introducing any user-provided values into a SQL query, care must be taken to avoid SQL injection attacks. SQL injection occurs when a program integrates a user-provided string value into a SQL query, and the user-provided value is crafted to terminate the string and perform another malicious SQL operation. To learn more about SQL injection, [see this page](/sql/relational-databases/security/sql-injection).
+>
+> The <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> and <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSqlInterpolated%2A> methods are safe against SQL injection, and always integrate parameter data as a separate SQL parameter. However, the <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSqlRaw%2A> method can be vulnerable to SQL injection attacks, if improperly used. See below for more details.
+
+The following example passes a single parameter to a stored procedure by including a parameter placeholder in the SQL query string and providing an additional argument:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureParameter)]
+
+While this syntax may look like regular C# [string interpolation](https://learn.microsoft.com/dotnet/csharp/language-reference/tokens/interpolated), the supplied value is wrapped in a `DbParameter` and the generated parameter name inserted where the `{0}` placeholder was specified. This makes > The <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> safe from SQL injection attacks, and sends the value efficiently and correctly to the database.
+
+When executing stored procedures, it can be useful to use named parameters in the SQL query string, especially when the stored procedure has optional parameters:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureNamedSqlParameter)]
+
+If you need more control over the database parameter being sent, you can also construct a `DbParameter` and supply it as a parameter value. This allows you to set the precise database type of the parameter, or facets such as its size, precision or length:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureSqlParameter)]
+
+> [!NOTE]
+>
+> Pay attention to the ordering of your parameters when executing a stored procedure. Unless using named parameters, the parameter in the SQL must exactly match the stored procedure definition, otherwise you may encounter type conversion exceptions and/or unexpected behavior when the procedure is executed.
```

Comment:
This feels like two notes; one for ordering, one for making sure the parameters match.

Reply:
You mean one note that ordering must match, and another that e.g. facets must match? If so, I'll make that clearer.